modules/OSSMailScanner/models/PrefixScannerAction.php

```diff
@@ -20,37 +20,44 @@ abstract class OSSMailScanner_PrefixScannerAction_Model
 
 	/**
 	 * Process.
+	 *
+	 * @param OSSMail_Mail_Model $mail
 	 */
 	abstract public function process(OSSMail_Mail_Model $mail);
 
 	/**
 	 * Find and bind.
 	 *
-	 * @return bool|int
+	 * @return bool|int|array
 	 */
 	public function findAndBind()
 	{
 		$mailId = $this->mail->getMailCrmId();
-		if (!$mailId) {
-			return 0;
-		}
-		$returnIds = [];
-		$query = (new \App\Db\Query())->select(['crmid'])->from('vtiger_ossmailview_relation')->where(['ossmailviewid' => $mailId]);
-		$dataReader = $query->createCommand()->query();
-		while ($crmId = $dataReader->readColumn(0)) {
-			if (\App\Record::getType($crmId) === $this->moduleName) {
-				$returnIds[] = $crmId;
+		$recordId = 0;
+		if ($mailId) {
+			$returnIds = [];
+			$query = (new \App\Db\Query())->select(['crmid'])->from('vtiger_ossmailview_relation')->where(['ossmailviewid' => $mailId]);
+			$dataReader = $query->createCommand()->query();
+			while ($crmId = $dataReader->readColumn(0)) {
+				if (\App\Record::getType($crmId) === $this->moduleName) {
```

Comment:
remove loop
add join to query and add condition [setype =>$this->moduleName, ['<>',deleted,1]]